Mark kins in KS test result output

the-x-at opened the pull request:
A new column `ksgrp` is added to the Kinship sum test, which groups affected individuals by kinship. Solves issue #15.

- [x]  `R CMD check` runs successfully.

the-x-at committed:
Mark kins in KS test result output

A new column `ksgrp` is added to the Kinship sum test, which groups
affected individuals by kinship. Solves issue #15.

jorainer commented on R/Methods-FAKinSumResults.R:
```diff
@@ -147,16 +149,45 @@ setMethod("result", "FAKinSumResults", function(object, method="BH"){
+##' IDs of the affected individuals. An NA indicates that the affected
+##' individual was above the threshold. The named vector corresponds to the
+##' entries in vector `padj`.
+define.result.groups.ks <- function(kin, padj, th=0.05)
```
should we maybe also add an additional parameter `minKinship = 0` to the function (and then also to `doShareKinship`) that allows to define which individuals to group? I guess we would not like to put e.g. 3rd cousins into the same "kin" group as their kin is only very low. The default could be still `0` but we could then also use a different kinship threshold - as I would like to not put, like said, third grade cousins into the same kinship group.

the-x-at replied:
Sure, that's easy.

jorainer replied:
Excellent! Please re-request my review once you've pushed to this PR and are OK with me reviewing it.

jorainer reviewed (changes requested):
This is an extremely helpfull additional functionality @the-x-at ! I have just one comment we should discuss before merging.

the-x-at committed:
Fix comparison to become TRUE again

A numeric comparison ended up evaluating to FALSE, even though the
results are expected to be equal. Rounding error.
In addition, my editor removed lots of trailing spaces.

the-x-at committed:
Add filter to remove remote relatives

- New parameter `rmKinship` to all `shareKinship` methods allows to
  remove distant relatives.
- Add unit test for these scenarios.
- Add it to all documentations.
- Add a short paragraph to the vignette.
- Tested with R 4.1.0 and Bioconductor 3.13.

the-x-at commented:
Was more work than I assumed. `shareKinship` exists in multiple locations. Documentation had to be updated. `R-4.1.0` was just released with Bioconductor 3.13, also tested it there. Overall, see my commit 7f2a396 for a full list of changes. I decided to call the new parameter `rmKinship`, as `minKinship` was misleading: it removes all kins with kinship ≤ `rmKinship`, whereas `minKinship` would indicate that the value itself would still be included.

jorainer reviewed (approved):
Looks good to me! Will fix the merge conflicts and CI and then merge.

jorainer committed:
Merge branch 'master' into ks-groups

jorainer committed:
Merge branch 'master' into ks-groups

jorainer committed:
Fix NEWS
